makefile

```diff
@@ -418,6 +425,9 @@ junit-report:
 	$(GO_TEST_CMD) bash -c "cat test_reports/*.txt | go-junit-report > test_reports/junit.xml"
 
 setup-proto: download
+	# Work around a "permission denied" error on macOS, when the following 'rm -rf' attempts to
```

Comment:
https://github.com/G-Research/armada/issues/1279 is the issue that is tracking these permission errors.  I'd like to fix this at the container level rather than sprinkling around these chmod everywhere.

Reply:
I mean, I agree this is kinda awful, but the bug itself is awful too. IMO, we shouldn't block this on #1279 given it represents literally a week of Rich's work.

Reply:
The chmod hack has been removed in commit f263b353 - I have borrowed the solution that @jayofdoom proposed in #1279 and it works consistently well now.

Reply:
https://app.circleci.com/pipelines/github/G-Research/armada/5799/workflows/4011e134-dcc0-4222-aa8d-7816b845d671/jobs/13592/parallel-runs/0/steps/0-104

You got a pretty serious fail in the integration tests.

Reply:
aha! This is how it broke in my PR too, I remember now.

Reply:
```
mkdir -p .build/server
docker run --rm -u 1001:1002 -v /home/circleci/go/src/github.com/G-Research/armada:/go/src/armada -w /go/src/armada --network=host -e GOPROXY -e GOPRIVATE -e INTEGRATION_ENABLED=true -e CGO_ENABLED=0 -e GOOS=linux -e GARCH=amd64 -v /home/circleci/.go_workspace:/go golang:1.16-buster go build -ldflags="-s -w" -o ./.build/server/server cmd/armada/main.go
cp -a ./config/armada ./.build/server/config
docker build  -t armada -f ./build/armada/Dockerfile ./.build/server/
failed to initialize build cache at /.cache/go-build: mkdir /.cache: permission denied
make: *** [makefile:225: build-docker-server] Error 1
```
What/where is this /.cache thing created?

Reply:
This code is what is failing:  

https://github.com/G-Research/armada/blob/master/makefile#L210

I guess its something to do with the docker cache?

Reply:
Looks like this may work but unsure what it does to build performance:

https://github.com/openshift/release/issues/9748

Reply:
Builds now appear to be working, both locally on my Macbook and Linux systems, and in GH Actions; please take a look, re-review, etc etc.

Reply:
Looks like there are still some issues on integration tests.

Reply:
We fixed this with a separate PR.